From d949b912eef7195e7e99cc02c7c1fdd7859350cf Mon Sep 17 00:00:00 2001
From: Chun-wei Fan <fanchunwei@src.gnome.org>
Date: Tue, 7 Jun 2016 15:51:31 +0800
Subject: [PATCH] config.h.win32.in: Always define HAVE_LONG_LONG

Visual Studio actually supports long long types, but HAVE_LONG_LONG is
undefined for Visual Studio builds, likely due to issues in previous
gnulib code for printf functionality, that was bundled with GLib.

Since gnulib has much better support with Visual Studio nowadays (which we
updated the related code to last October), and HAVE_LONG_LONG being undefined
actually causes issues in Visual Studio builds, which was demonstrated with
the type-test test program in tests/, we should always define HAVE_LONG_LONG
in config.h.win32.in.

Thanks to Paolo Borelli for the heads up on the issue.
---
 config.h.win32 | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/config.h.win32 b/config.h.win32
index 1da7bd3..b1c69df 100644
--- a/config.h.win32
+++ b/config.h.win32
@@ -288,11 +288,7 @@
 #define HAVE_LONG_DOUBLE 1
 
 /* Define if you have the 'long long' type. */
-#ifndef _MSC_VER
 #define HAVE_LONG_LONG 1
-#else /* _MSC_VER */
-/* #undef HAVE_LONG_LONG */
-#endif /* _MSC_VER */
 
 /* define if system printf can print long long */
 #define HAVE_LONG_LONG_FORMAT 1
-- 
2.7.4

